compiler/element/backend/envoy_native/appnettype.py

```diff
@@ -0,0 +1,116 @@
+# Created and used by Envoy natvie backend.
+
+
+from typing import Optional
+from compiler.element.logger import ELEMENT_LOG as LOG
+
+from compiler.element.backend.envoy_native.types import *
+
+
+
+# Every expression, state and temporary variables in AppNet belong to a AppNetType. 
```

Comment:
Do you mean that this file is used for type inference? In this case, I suggest that it should be moved to the frontend/ir folder as it's backend-independent.

Reply:
Yes. But changing fronend/ir may cause problems in other backends, so I keep it in the native backend for now. It should be moved outside in future.